src/historywork/PutSnapshotFilesWork.cpp

```diff
@@ -121,6 +122,12 @@ PutSnapshotFilesWork::getFilesToZip()
              mSnapshot->differingHASFiles(getState->getHistoryArchiveState()))
         {
             filesToZip.insert(f->localPath_nogz());
+            // Bucket files are tracked outside of the temp folder, so we
+            // manually collect buckets to clean up at the end
+            if (BucketManager::isBucketFile(f->baseName_gz()))
+            {
+                bucketsToCleanup.insert(f->localPath_gz());
```

Comment:
it would be cleaner to return a `UnorderedMap<std::string, FileTransferInfo>` instead of `UnorderedSet<std::string>` and just store it in a member variable.

That way, your cleanup function can just delete all `localPath_gz()` and `localPath_gz_tmp()` associated with those. I don't think you need to only delete buckets: any file that got zipped can safely be deleted.

Reply:
other files are deleted automatically because they live in temp directories, so I don't think we need to do this.

Reply:
we shouldn't need to delete `tmp` files here. Those are transient and owned by the unzip work. If they are left, that would indicate there's a bug in unzip work, that it's not cleaned up correctly (I verified that `GetAndUnzipRemoteFileWork` cleans up as expected).

Reply:
I think of the "zip files" (all of them) as resources that this work "inherits" on success (`GzipWork` on success stops tracking files) so it should be cleaning up after itself.

The fact that other files are downloaded in temp folders is a choice made somewhere elsee. The fact that they happen to be deleted by the caller is more of a "catch all" type of situation that happens to cleanup dangling files. The current work is the one that owns those files.

I would err on writing the simplest code possible here unless there are disadvantages.

Adding to this, I am not sure we gain much by trying to not delete files in the abort/failure case, which is why I was saying to just delete everything regardless of success/failure/abort.

We could *not* delete the tmp files, but that implies (I think) fixing the abort path. I would recommend deleting them for now, and fix abort in a subsequent PR.